src/Value/MessageBatch.php

```diff
@@ -0,0 +1,41 @@
+<?php
+
+declare(strict_types=1);
+
+namespace KafkaRest\Value;
+
+use function count;
+
+final class MessageBatch
+{
+    /** @var array<Message> */
+    private $messages;
+
+    public function add(Message $message) : void
+    {
+        $this->messages[] = $message;
+    }
+
+    /**
+     * @return array<Message>
+     */
+    public function getMessages() : array
+    {
+        return $this->messages;
+    }
+
+    public function count() : int
+    {
+        return count($this->messages);
+    }
+
+    public function isEmpty() : bool
+    {
+        return count($this->messages) === 0;
+    }
+
+    public function getLast() : Message
+    {
+        return $this->messages[count($this->messages) - 1];
```

Comment:
ban php7.2 and use array_key_last? 😏

Reply:
I don't think it's necessary, not because of this one function...